src/sync/sync_permission.hpp

```diff
@@ -93,12 +93,24 @@ class PermissionResults {
         return m_results.async(std::move(target));
     }
 
-    // Create a new instance by further filtering or sorting this instance.
+    // Create a new instance by further filtering this instance.
     PermissionResults filter(Query&& q) const
     {
         return PermissionResults(m_results.filter(std::move(q)));
     }
 
+    // Create a new instance by sorting this instance.
+    PermissionResults sort(SortDescriptor&& s) const
+    {
+        return PermissionResults(m_results.sort(std::move(s)));
+    }
+
+    // Get the results.
+    Results& results()
```

Comment:
From that pr it looks like all you really need is the `realm` and `objectSchema`, but not direct access to the actual results. I'm also not a fan of looking up column indexes with core apis if the value is already cached in the objectSchema (unless this isn't actually the case?)